src/System.Net.Http/src/System/Net/Http/Managed/HttpEnvironmentProxy.cs

```diff
@@ -0,0 +1,265 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Net.Http;
+using System.Net;
+using System.Collections.Generic;
+
+namespace System.Net.Http
+{
+    public sealed class HttpEnvironmentProxyCredientials : ICredentials
+    {
+        // Wrapper class for cases when http and https has different authentication.
+        private NetworkCredential _http;
+        private NetworkCredential _https;
+
+        public HttpEnvironmentProxyCredientials(NetworkCredential http, NetworkCredential https)
+        {
+            _http = http;
+            _https = https;
+        }
+
+        public NetworkCredential GetCredential(Uri uri, string authType)
+        {
+            return uri.Scheme == Uri.UriSchemeHttp ? _http : _https;
+        }
+    }
+
+    internal sealed class HttpEnvironmentProxy : IWebProxy
+    {
+        private const string envAllProxyUC = "ALL_PROXY";
+        private const string envAllProxyLC = "all_proxy";
```

Comment:
Formatting guidelines, we use PascalCasing for all consts, https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md